src/test/java/org/geppetto/simulator/external/tests/NeuronNSGSimulatorServiceTest.java

```diff
@@ -0,0 +1,174 @@
+ * 
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the MIT License
+ * which accompanies this distribution, and is available at
+ * http://opensource.org/licenses/MIT
+ *
+ * Contributors:
+ *     	OpenWorm - http://openworm.org/people.html
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights 
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell 
+ * copies of the Software, and to permit persons to whom the Software is 
+ * furnished to do so, subject to the following conditions:
+ * 
+ * The above copyright notice and this permission notice shall be included in 
+ * all copies or substantial portions of the Software.
+ * 
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR 
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, 
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. 
+ * IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, 
+ * DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR 
+ * OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE 
+ * USE OR OTHER DEALINGS IN THE SOFTWARE.
+ *******************************************************************************/
+package org.geppetto.simulator.external.tests;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+
+import junit.framework.Assert;
+
+import org.geppetto.core.beans.SimulatorConfig;
+import org.geppetto.core.common.GeppettoExecutionException;
+import org.geppetto.core.common.GeppettoInitializationException;
+import org.geppetto.core.data.model.IAspectConfiguration;
+import org.geppetto.core.data.model.ResultsFormat;
+import org.geppetto.core.services.registry.ServicesRegistry;
+import org.geppetto.core.simulation.ISimulatorCallbackListener;
+import org.geppetto.core.simulator.RemoteSimulatorConfig;
+import org.geppetto.model.ExternalDomainModel;
+import org.geppetto.model.GeppettoFactory;
+import org.geppetto.simulator.remote.services.NSGUtilities;
+import org.geppetto.simulator.remote.services.NeuronNSGSimulatorService;
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.ngbw.directclient.CiCipresException;
+
+public class NeuronNSGSimulatorServiceTest implements ISimulatorCallbackListener
+{
+
+	//AQP Review all these libs
+	private static String dirToExecute;
+	private static String fileToExecute;
+	private static NeuronNSGSimulatorService simulator;
+	private static String resultsDir;
+	private static boolean done = false;
+
+	@BeforeClass
+	public static void setup()
+	{
+//		File dir = new File(NeuronNSGSimulatorServiceTest.class.getResource("/neuronConvertedModel/").getFile());
```

Comment:
@tarelli Yes, for the moment. Otherwise, a username/password will be required for anybody that wants to use the simulator.external bundle. I haven't deleted them as it is useful for me in order to do some testing. If you don't mind, I prefer to keep them in the code, we can think about a way to test the simulator in the future... if it is too dirty for you let me know, I will delete them and keep them locally.